gobblin-runtime/src/main/java/org/apache/gobblin/runtime/AbstractJobLauncher.java

```diff
@@ -439,6 +450,7 @@ public void launchJob(JobListener jobListener)
       TimingEvent launchJobTimer = this.eventSubmitter.getTimingEvent(TimingEvent.LauncherTimings.FULL_JOB_EXECUTION);
 
       try (Closer closer = Closer.create()) {
+        addShutdownHook(closer);
```

Comment:
what's the shutdown hook doing here? Seems like the closer is created in the try clause which should be automatically closed? Am I missing any thing here?

Reply:
the closer that is created in `try` is not closed when there is a kill. `finally` blocks does not run when the jvm is abruptly stopping. Adding a shutdown hook makes sure that close is called in those cases.